### PR TITLE
update: update spark version to 3.1.2

### PR DESCRIPTION
**Pull request by xuwq1993**
Update Spark Version to 3.1.2 (Synapse Spark3 GA version)

### CODE REVIEW

File: src/main/scala/com/microsoft/ml/spark/vw/VowpalWabbitContextualBandit.scala [hunk 1]
```diff
@@ -275,7 +275,7 @@ class VowpalWabbitContextualBandit(override val uid: String)
     })
   }
 
-  override def fit(dataset: Dataset[_], paramMaps: Array[ParamMap]): Seq[VowpalWabbitContextualBanditModel] = {
```
Comment (eisber) [inline src/main/scala/com/microsoft/ml/spark/vw/VowpalWabbitContextualBandit.scala:278]: why is the override removed?

**Review by eisber — APPROVED** [review]

File: src/main/scala/com/microsoft/ml/spark/vw/VowpalWabbitContextualBandit.scala (hunk 1 above)
Comment (xuwq1993) [reply src/main/scala/com/microsoft/ml/spark/vw/VowpalWabbitContextualBandit.scala:278]: Spark ML Lib changed Array[ParamMap] to Seq[ParamMap]
Update: keep "override", changed Array[] to Seq[]

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**